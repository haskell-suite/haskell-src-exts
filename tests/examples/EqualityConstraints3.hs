{-# LANGUAGE GADTs #-}

one :: (~) a Int => a
one = 1
