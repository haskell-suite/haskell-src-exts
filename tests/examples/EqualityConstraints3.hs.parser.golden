ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan =
            SrcSpan "tests/examples/EqualityConstraints3.hs" 1 1 5 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/EqualityConstraints3.hs" 1 1 1 1
            , SrcSpan "tests/examples/EqualityConstraints3.hs" 3 1 3 1
            , SrcSpan "tests/examples/EqualityConstraints3.hs" 3 1 3 1
            , SrcSpan "tests/examples/EqualityConstraints3.hs" 3 1 3 1
            , SrcSpan "tests/examples/EqualityConstraints3.hs" 4 1 4 1
            , SrcSpan "tests/examples/EqualityConstraints3.hs" 5 1 5 1
            , SrcSpan "tests/examples/EqualityConstraints3.hs" 5 1 5 1
            ]
        }
      Nothing
      [ LanguagePragma
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/EqualityConstraints3.hs" 1 1 1 41
            , srcInfoPoints =
                [ SrcSpan "tests/examples/EqualityConstraints3.hs" 1 1 1 13
                , SrcSpan "tests/examples/EqualityConstraints3.hs" 1 19 1 20
                , SrcSpan "tests/examples/EqualityConstraints3.hs" 1 38 1 41
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/EqualityConstraints3.hs" 1 14 1 19
                , srcInfoPoints = []
                }
              "GADTs"
          , Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/EqualityConstraints3.hs" 1 21 1 37
                , srcInfoPoints = []
                }
              "FlexibleContexts"
          ]
      ]
      []
      [ TypeSig
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/EqualityConstraints3.hs" 3 1 3 22
            , srcInfoPoints =
                [ SrcSpan "tests/examples/EqualityConstraints3.hs" 3 5 3 7 ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/EqualityConstraints3.hs" 3 1 3 4
                , srcInfoPoints = []
                }
              "one"
          ]
          (TyForall
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/EqualityConstraints3.hs" 3 8 3 22
               , srcInfoPoints = []
               }
             Nothing
             (Just
                (CxSingle
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/EqualityConstraints3.hs" 3 8 3 20
                     , srcInfoPoints =
                         [ SrcSpan "tests/examples/EqualityConstraints3.hs" 3 18 3 20 ]
                     }
                   (ClassA
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/EqualityConstraints3.hs" 3 8 3 20
                        , srcInfoPoints =
                            [ SrcSpan "tests/examples/EqualityConstraints3.hs" 3 18 3 20 ]
                        }
                      (Special
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/EqualityConstraints3.hs" 3 8 3 11
                           , srcInfoPoints =
                               [ SrcSpan "tests/examples/EqualityConstraints3.hs" 3 8 3 9
                               , SrcSpan "tests/examples/EqualityConstraints3.hs" 3 9 3 10
                               , SrcSpan "tests/examples/EqualityConstraints3.hs" 3 10 3 11
                               ]
                           }
                         (TyEqCon
                            SrcSpanInfo
                              { srcInfoSpan =
                                  SrcSpan "tests/examples/EqualityConstraints3.hs" 3 8 3 11
                              , srcInfoPoints =
                                  [ SrcSpan "tests/examples/EqualityConstraints3.hs" 3 8 3 9
                                  , SrcSpan "tests/examples/EqualityConstraints3.hs" 3 9 3 10
                                  , SrcSpan "tests/examples/EqualityConstraints3.hs" 3 10 3 11
                                  ]
                              }))
                      [ TyVar
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/EqualityConstraints3.hs" 3 12 3 13
                            , srcInfoPoints = []
                            }
                          (Ident
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan "tests/examples/EqualityConstraints3.hs" 3 12 3 13
                               , srcInfoPoints = []
                               }
                             "a")
                      , TyCon
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/EqualityConstraints3.hs" 3 14 3 17
                            , srcInfoPoints = []
                            }
                          (UnQual
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan "tests/examples/EqualityConstraints3.hs" 3 14 3 17
                               , srcInfoPoints = []
                               }
                             (Ident
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan "tests/examples/EqualityConstraints3.hs" 3 14 3 17
                                  , srcInfoPoints = []
                                  }
                                "Int"))
                      ])))
             (TyVar
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/EqualityConstraints3.hs" 3 21 3 22
                  , srcInfoPoints = []
                  }
                (Ident
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/EqualityConstraints3.hs" 3 21 3 22
                     , srcInfoPoints = []
                     }
                   "a")))
      , PatBind
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/EqualityConstraints3.hs" 4 1 4 8
            , srcInfoPoints = []
            }
          (PVar
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/EqualityConstraints3.hs" 4 1 4 4
               , srcInfoPoints = []
               }
             (Ident
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/EqualityConstraints3.hs" 4 1 4 4
                  , srcInfoPoints = []
                  }
                "one"))
          (UnGuardedRhs
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/EqualityConstraints3.hs" 4 5 4 8
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/EqualityConstraints3.hs" 4 5 4 6 ]
               }
             (Lit
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/EqualityConstraints3.hs" 4 7 4 8
                  , srcInfoPoints = []
                  }
                (Int
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/EqualityConstraints3.hs" 4 7 4 8
                     , srcInfoPoints = []
                     }
                   1
                   "1")))
          Nothing
      ]
  , []
  )
